source/Uniform/InMemory/InMemoryCollection.cs

```diff
@@ -18,12 +16,12 @@ public class InMemoryCollection : IDocumentCollection, IInMemoryCollection
         /// <summary>
         /// Main data structure that contains all documents of type TDocument, hashed by key
         /// </summary>
-        private readonly Dictionary<String, Object> _documents = new Dictionary<String, Object>();
+        private readonly ConcurrentDictionary<String, Object> _documents = new ConcurrentDictionary<String, Object>();

```

Comment:
I don't think it is a good idea to use ConcurrentDictionary here. It slows down significantly read/write access to it. Even in single threaded tests with setting 3000000 elements it shows 4 times slower results rather then Dictionary. Situation will be worth in concurrent tests. I suppose it is bad idea of try to support threadsafe libraries. If you are fighting for performance - use "share nothing" approach.